### DISCUSSION TITLE
docs for filters or predicates

### DISCUSSION THREAD

**Discussion by dongho-jung** [bpftrace/bpftrace#1995] [Q&A] [answered]
according to man pages
```
   PREDICATES
       Define conditions for which a probe should be executed:

       kprobe:sys_open / uid == 0 / { ... }
```
as above said, I can filter them like `foo == bar`. But is that all? I wonder can it achieve other things like `foo > bar` or `foo in (bar, baz)` or `foo == bar or foo != baz`.

I couldn't find the deeper reference for this. or I've missed.

thanks

**Comment by mmisono** [comment, marked as answer]
You can write a condition in a filter in the same way as if statements. `/ $foo > $bar /` is a valid expression (assuming that `foo` and `bar` are variables). You can write `/ @bar[$foo] || @baz[$foo] /` to check if BPF maps has a (non-zero) value with the specified key.

**Reply by dongho-jung** [reply]
oh associative array, I've missed that. Now I can check whethe it's included or not. Thanks!
